sdk/core/src/pipeline.rs

```diff
@@ -79,8 +80,36 @@ where
         // TODO: Add transport policy for WASM once https://github.com/Azure/azure-sdk-for-rust/issues/293 is resolved.
         #[cfg(not(target_arch = "wasm32"))]
         {
-            let transport_policy = TransportPolicy::new(&options.transport);
-            pipeline.push(Arc::new(transport_policy));
+            // This code replaces the default transport policy at runtime if these two conditions
+            // are met:
+            // 1. The mock_transport_framework is enabled
+            // 2. The environmental variable TESTING_MODE is either RECORD or PLAY
+            #[cfg(not(feature = "mock_transport_framework"))]
+            pipeline.push(Arc::new(TransportPolicy::new(options.transport.clone())));
+
+            #[cfg(feature = "mock_transport_framework")]
+            match &std::env::vars()
```

Comment:
Why not use `std::env::var`?

Reply:
I wanted it to be case insensitive. Will fix.